com.unity.shadergraph/Editor/Drawing/Controllers/BlackboardController.cs

```diff
@@ -1,21 +1,478 @@
+    }
+
+    class BlackboardController : SGViewController<GraphData, BlackboardViewModel>
+    {
+        // Type changes (adds/removes of Types) only happen after a full assmebly reload so its safe to make this stuff static (I think, confirm this assumption)
+        static IList<Type> m_shaderInputTypes;
+
+        static BlackboardController()
+        {
+            var shaderInputTypes = TypeCache.GetTypesWithAttribute<BlackboardInputInfo>().ToList();
+            // Sort the ShaderInput by priority using the BlackboardInputInfo attribute
+            shaderInputTypes.Sort((s1, s2) => {
+                var info1 = Attribute.GetCustomAttribute(s1, typeof(BlackboardInputInfo)) as BlackboardInputInfo;
+                var info2 = Attribute.GetCustomAttribute(s2, typeof(BlackboardInputInfo)) as BlackboardInputInfo;
+
+                if (info1.priority == info2.priority)
+                    return (info1.name ?? s1.Name).CompareTo(info2.name ?? s2.Name);
+                else
+                    return info1.priority.CompareTo(info2.priority);
+            });
+
+            m_shaderInputTypes = shaderInputTypes.ToList();
+        }
+
+        internal int k_PropertySectionIndex = 0;
+        internal int k_KeywordSectionIndex = 1;
+
+        BlackboardSectionController m_PropertySectionController;
+        BlackboardSectionController m_KeywordSectionController;
+        IList<BlackboardSectionController> m_BlackboardSectionControllers = new List<BlackboardSectionController>();
+
+        SGBlackboard m_Blackboard;
+
+        internal SGBlackboard blackboard
+        {
+            get => m_Blackboard;
+            private set => m_Blackboard = value;
+        }
+
+        void InitializeViewModel()
+        {
+            // Clear the view model
+            ViewModel.Reset();
+
+            ViewModel.subtitle = BlackboardUtils.FormatPath(Model.path);
+
+            // TODO: Could all this below data be static in the view model as well? Can't really see it ever changing at runtime
+            // All of this stuff seems static and driven by attributes so it would seem to be compile-time defined, could definitely be an option
+            // Only issue is checking for conflicts with user made keywords, could leave just that bit here and make everything else static
+
+            // Property data first
+            foreach (var shaderInputType in m_shaderInputTypes)
+            {
+                if (shaderInputType.IsAbstract)
+                    continue;
+
+                var info = Attribute.GetCustomAttribute(shaderInputType, typeof(BlackboardInputInfo)) as BlackboardInputInfo;
+                string name = info?.name ?? ObjectNames.NicifyVariableName(shaderInputType.Name.Replace("ShaderProperty", ""));
+
+                // QUICK FIX TO DEAL WITH DEPRECATED COLOR PROPERTY
+                if (name.Equals("Color", StringComparison.InvariantCultureIgnoreCase) && ShaderGraphPreferences.allowDeprecatedBehaviors)
```

Comment:
We want deprecated _and_ new versions in this case, right? Looks like this code will only add the deprecated version.

Reply:
Yes good catch, for whatever reason I thought it was an either/or behavior desired here as opposed to, you can create the deprecated Colors AND create the new proper colors, when that flag is enabled. Will change it so both get added.